enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,258 @@
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The baremetal platform is switching from a QCOW2 format for its CoreOS image to
+an ISO. To ensure that existing disconnected clusters can update to this, the
+ISO image will be included in the release payload. This will be balanced out by
+removing the RHEL image currently shipped in the release payload. There are
+many other nice benefits, such that after future upgrades clusters will
+provision new machines with the latest CoreOS, without requiring an extra
+reboot.
+
+## Motivation
+
+Currently, the deploy disk image (i.e. the image running IPA -
+ironic-python-agent) is a RHEL kernel plus initrd that is installed (from an
+RPM) into the `ironic-ipa-downloader` container image, which in turn is part of
+the OpenShift release payload. When the metal3 Pod starts up, the disk image is
+copied from the container to a HostPath volume whence it is available to
+Ironic.
+
+The provisioning OS disk image is a separate CoreOS QCOW2 image. The URL for
+this is known by the installer. It points to the cloud by default and may be
+customised by the user to allow disconnected installs. The URL is stored in the
+Provisioning CR at install time and never updated automatically. The image
+itself is downloaded once and permanently cached on all of the master nodes.
+Never updating the image is tolerable because, upon booting, the CoreOS image
+will update itself to the version matching the cluster it is to join. It
+remains suboptimal because new Machines will take longer and longer (and more
+and more bandwidth) to join as the cluster ages. This issue exists on all
+platforms, and is the subject of a [long-standing enhancement
+proposal](https://github.com/openshift/enhancements/pull/201). Other issues
+specific to the baremetal platform are that boot times for bare metal servers
+can be very long (and therefore the reboot is costly), and that support for
+particular hardware may theoretically require a particular version of CoreOS.
+
+We are changing the deploy disk image to use CoreOS. This takes the form of
+both an ISO (for hosts that can use virtualmedia) and of a kernel + initrd +
+rootfs (for hosts that use PXE). When upgrading an existing disconnected
+cluster, we have no way to acquire these images without the user manually
+intervening to mirror them.
+
+Like the QCOW2 provisioning disk image, the URLs for these images are known by
+the installer, but they point to the cloud by default and would have to be
+customised by the user at install time to allow disconnected installs. IPA
+itself is delivered separately, as a container image as part of the OpenShift
+release payload.
```

Comment:
The terminology here (and above) could perhaps be confusing, e.g "provisioning disk image" vs "deploy disk image" - maybe say "target disk image" or "provisioned disk image" for the provisioning one, so it's clear that's what we're writing to disk, vs using during provisioning?

I'd also perhaps stick to saying e.g "The deploy disk image itself is delivered separately ..." for the same reason - unless you are familiar with IPA it's not super-clear we're talking about the deploy image when saying "IPA" here?

Reply:
> I'd also perhaps stick to saying e.g "The deploy disk image itself is delivered separately ..." for the same reason - unless you are familiar with IPA it's not super-clear we're talking about the deploy image when saying "IPA" here?

We're not :) What this is saying that IPA is delivered separately to the deploy image, so that (next paragraph) even if you use an out-of-date CoreOS, you will still pull the latest IPA from the registry in the cluster, since it's part of the release payload. I'll rearrange this so it doesn't jump around so much.

IPA and "deploy image" are defined at the top of this section.